sdk/core/Azure.Core/perf/SerializationBenchmark/SerializationBenchmark.cs

```diff
@@ -0,0 +1,230 @@
+            _options = ModelSerializerOptions.AzureServiceDefault;
+            _content = new SequenceWriter();
+            using Utf8JsonWriter writer = new Utf8JsonWriter(_content);
+            _model.Serialize(writer, new ModelSerializerOptions());
+            writer.Flush();
+            _sequence = _content.GetReadOnlySequence();
+            _buffer = new byte[2000];
+        }
+
+        [Benchmark]
+        [BenchmarkCategory("Internal")]
+        public void Serialize_Internal()
+        {
+            using var stream = new MemoryStream();
+            using var writer = new Utf8JsonWriter(stream);
+            Serialize(writer);
+            writer.Flush();
+        }
+
+        [Benchmark]
+        [BenchmarkCategory("Cast")]
+        public void Serialize_ImplicitCast()
+        {
+            using var x = CastToRequestContent();
+        }
+
+        [Benchmark]
+        [BenchmarkCategory("Cast")]
+        public bool Serialize_ImplicitCastWithSerialize()
+        {
+            using var x = CastToRequestContent();
+            return x.TryComputeLength(out var length);
+        }
+
+        [Benchmark]
+        [BenchmarkCategory("Cast")]
+        public void Serialize_ImplicitCastWithUsage()
+        {
+            using var x = CastToRequestContent();
+            x.TryComputeLength(out var length);
+            using var stream = new MemoryStream((int)length);
+            x.WriteTo(stream, default);
+        }
+
+        [Benchmark]
+        [BenchmarkCategory("Cast")]
+        public RequestContent CreateRequestContent()
+        {
+            return RequestContent.Create(_content);
+        }
+
+        [Benchmark]
+        [BenchmarkCategory("ModelJsonConverter")]
+        public string Serialize_ModelJsonConverter()
+        {
+            JsonSerializerOptions options = new JsonSerializerOptions();
+            options.Converters.Add(new ModelJsonConverter(ModelSerializerFormat.Wire));
+            return JsonSerializer.Serialize(_model, options);
+        }
+
+        [Benchmark]
+        [BenchmarkCategory("ModelSerializer")]
+        public BinaryData Serialize_ModelSerializer()
+        {
+            return ModelSerializer.Serialize(_model, _options);
+        }
+
+        [Benchmark]
+        [BenchmarkCategory("ModelSerializer")]
+        public BinaryData Serialize_ModelSerializerNonGeneric()
+        {
+            return ModelSerializer.Serialize((object)_model, _options);
+        }
+
+        [Benchmark]
+        [BenchmarkCategory("PublicInterface")]
+        public void Serialize_PublicInterface()
+        {
+            using var content = new SequenceWriter();
+            using var writer = new Utf8JsonWriter(content);
+            _model.Serialize(writer, _options);
+            writer.Flush();
+        }
+
+        [Benchmark]
+        [BenchmarkCategory("Internal")]
+        public T Deserialize_Internal()
+        {
+            using JsonDocument doc = JsonDocument.Parse(_json);
```

Comment:
It looks like this is parsing from a string.  Since we get UTF8 JSON bytes from the wire, it might be more efficient to parse from that directly, to avoid the transcoding to UTF16 step.  I'm going to try to do this in the JSON Merge Patch PR to see if I can optimize by avoiding transcoding - I'll let you know how it goes.

Reply:
Oh wait, I see you have the other code path as well.  Feel free to ignore.

Reply:
I moved this to set up to measure just the deserialization